src/main/java/com/appdynamics/extensions/webspheremq/metricscollector/ReadConfigurationEventQueueCollector.java

```diff
@@ -0,0 +1,175 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.appdynamics.extensions.webspheremq.metricscollector;
+
+import com.appdynamics.extensions.MetricWriteHelper;
+import com.appdynamics.extensions.conf.MonitorContextConfiguration;
+import com.appdynamics.extensions.logging.ExtensionsLoggerFactory;
+import com.appdynamics.extensions.metrics.Metric;
+import com.appdynamics.extensions.webspheremq.common.WMQUtil;
+import com.appdynamics.extensions.webspheremq.config.QueueManager;
+import com.appdynamics.extensions.webspheremq.config.WMQMetricOverride;
+import com.google.common.collect.Lists;
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQGetMessageOptions;
+import com.ibm.mq.MQMessage;
+import com.ibm.mq.MQQueue;
+import com.ibm.mq.MQQueueManager;
+import com.ibm.mq.constants.CMQC;
+import com.ibm.mq.constants.CMQCFC;
+import com.ibm.mq.constants.MQConstants;
+import com.ibm.mq.headers.pcf.MQCFH;
+import com.ibm.mq.headers.pcf.PCFMessage;
+import com.ibm.mq.headers.pcf.PCFMessageAgent;
+import com.singularity.ee.agent.systemagent.api.exception.TaskExecutionException;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+public class ReadConfigurationEventQueueCollector implements MetricsPublisher {
+
+	private static final Logger logger = ExtensionsLoggerFactory.getLogger(ReadConfigurationEventQueueCollector.class);
+	private final CountDownLatch countDownLatch;
+	private final MetricWriteHelper metricWriteHelper;
+	private final QueueManager queueManager;
+	private final PCFMessageAgent agent;
+	private final MonitorContextConfiguration monitorContextConfig;
+	private final MQQueueManager mqQueueManager;
+	private final Map<String, WMQMetricOverride> metricsToReport;
+	private final long bootTime;
+
+	public ReadConfigurationEventQueueCollector(Map<String, WMQMetricOverride> metricsToReport, MonitorContextConfiguration monitorContextConfig, PCFMessageAgent agent, MQQueueManager mqQueueManager, QueueManager queueManager, MetricWriteHelper metricWriteHelper, CountDownLatch countDownLatch) {
+		this.metricsToReport = metricsToReport;
+		this.monitorContextConfig = monitorContextConfig;
+		this.agent = agent;
+		this.mqQueueManager = mqQueueManager;
+		this.queueManager = queueManager;
+		this.metricWriteHelper = metricWriteHelper;
+		this.countDownLatch = countDownLatch;
+		this.bootTime = System.currentTimeMillis();
+	}
```

Comment:
The countdownLatch isn't used by this class.

```suggestion
	private static final Logger logger = ExtensionsLoggerFactory.getLogger(ReadConfigurationEventQueueCollector.class);
	private final MetricWriteHelper metricWriteHelper;
	private final QueueManager queueManager;
	private final PCFMessageAgent agent;
	private final MonitorContextConfiguration monitorContextConfig;
	private final MQQueueManager mqQueueManager;
	private final Map<String, WMQMetricOverride> metricsToReport;
	private final long bootTime;

	public ReadConfigurationEventQueueCollector(Map<String, WMQMetricOverride> metricsToReport, MonitorContextConfiguration monitorContextConfig, PCFMessageAgent agent, MQQueueManager mqQueueManager, QueueManager queueManager, MetricWriteHelper metricWriteHelper) {
		this.metricsToReport = metricsToReport;
		this.monitorContextConfig = monitorContextConfig;
		this.agent = agent;
		this.mqQueueManager = mqQueueManager;
		this.queueManager = queueManager;
		this.metricWriteHelper = metricWriteHelper;
		this.bootTime = System.currentTimeMillis();
	}
```